src/rapidata/rapidata_client/datapoints/_datapoint_uploader.py

```diff
@@ -0,0 +1,129 @@
+from rapidata.api_client.models.text_asset_input import TextAssetInput
+from rapidata.rapidata_client.datapoints._datapoint import Datapoint
+from rapidata.service.openapi_service import OpenAPIService
+from rapidata.api_client.models.multi_asset_input_assets_inner import (
+    MultiAssetInput,
+    MultiAssetInputAssetsInner,
+)
+from rapidata.api_client.models.create_datapoint_model import CreateDatapointModel
+from rapidata.api_client.models.create_datapoint_model_asset import (
+    CreateDatapointModelAsset,
+)
+from rapidata.api_client.models.create_datapoint_result import CreateDatapointResult
+from rapidata.api_client.models.create_datapoint_from_files_model_metadata_inner import (
+    CreateDatapointFromFilesModelMetadataInner,
+)
+from rapidata.api_client.models.existing_asset_input import ExistingAssetInput
+from rapidata.rapidata_client.datapoints._asset_uploader import AssetUploader
+from rapidata.rapidata_client.datapoints.metadata import (
+    PromptMetadata,
+    MediaAssetMetadata,
+    PrivateTextMetadata,
+    SelectWordsMetadata,
+    Metadata,
+)
+
+
+class DatapointUploader:
+    def __init__(self, openapi_service: OpenAPIService):
+        self.openapi_service = openapi_service
+        self.asset_uploader = AssetUploader(openapi_service)
+
+    def upload_datapoint(
+        self, datapoint: Datapoint, dataset_id: str, index: int
+    ) -> CreateDatapointResult:
+        metadata = self._get_metadata(datapoint)
+
+        uploaded_asset = (
+            self._handle_media_datapoint(datapoint)
+            if datapoint.data_type == "media"
+            else self._handle_text_datapoint(datapoint)
+        )
+        return self.openapi_service.dataset_api.dataset_dataset_id_datapoint_post(
+            dataset_id=dataset_id,
+            create_datapoint_model=CreateDatapointModel(
+                asset=uploaded_asset,
+                metadata=metadata,
```

Comment:
why are you retrieving the metadata to reupload it? the backend is ensuring that the metadata of the asset are stored on the datapoint aswell, only provide additional metadata, like the prompt and such